go/vt/vttablet/tabletserver/schema/historian.go

```diff
@@ -0,0 +1,286 @@
+	if err := proto.Unmarshal(row[4].ToBytes(), sch); err != nil {
+		return nil, 0, err
+	}
+	log.Infof("Read tracked schema from db: id %d, pos %v, ddl %s, schema len %d, time_updated %d \n",
+		id, mysql.EncodePosition(pos), ddl, len(sch.Tables), timeUpdated)
+
+	tables := map[string]*binlogdatapb.MinimalTable{}
+	for _, t := range sch.Tables {
+		tables[t.Name] = t
+	}
+	trackedSchema := &TrackedSchema{
+		schema: tables,
+		pos:    pos,
+		ddl:    ddl,
+	}
+	return trackedSchema, id, nil
+}
+
+// loadFromDB loads all rows from the schema_version table that the historian does not have as yet
+func (h *HistorianSvc) loadFromDB(ctx context.Context) error {
+	conn, err := h.he.GetConnection(ctx)
+	if err != nil {
+		return err
+	}
+	defer conn.Recycle()
+	log.Infof("In loadFromDb for lastID %d", h.lastID)
+	tableData, err := conn.Exec(ctx, fmt.Sprintf(getSchemaVersions, h.lastID), 10000, true)
+	if err != nil {
+		log.Infof("Error reading schema_tracking table %v", err)
+		return nil
+	}
+	log.Infof("Received rows from schema_version: %v", len(tableData.Rows))
+	if len(tableData.Rows) > 0 {
+		for _, row := range tableData.Rows {
+			trackedSchema, id, err := h.readRow(row)
+			if err != nil {
+				return err
+			}
+			h.schemas = append(h.schemas, trackedSchema)
+			h.lastID = id
+		}
+		h.sortSchemas()
+	}
+	return nil
+}
+
+// Open opens the underlying schema Engine. Called directly by a user purely interested in schema.Engine functionality
+func (h *HistorianSvc) Open() error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	if h.isOpen {
+		return nil
+	}
+	h.he.Open()
+	ctx := tabletenv.LocalContext()
+	h.reload(ctx)
+	if err := h.loadFromDB(ctx); err != nil {
+		return err
+	}
+	h.isOpen = true
+	return nil
+}
+
+// Close closes the underlying schema engine and empties the version cache
+func (h *HistorianSvc) Close() {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	if !h.isOpen {
+		return
+	}
+	h.schemas = nil
+	h.he.Close()
+	h.isOpen = false
+}
+
+// reload gets the latest schema and replaces the latest copy of the schema maintained by the historian
+func (h *HistorianSvc) reload(ctx context.Context) error {
+	log.Info("In historian reload")
+	if err := h.he.Reload(ctx); err != nil {
+		return err
+	}
+	tables := map[string]*binlogdatapb.MinimalTable{}
+	log.Infof("Schema returned by engine has %d tables", len(h.he.GetSchema()))
+	for _, t := range h.he.GetSchema() {
+		table := &binlogdatapb.MinimalTable{
+			Name:   t.Name.String(),
+			Fields: t.Fields,
+		}
+		var pkc []int64
+		for _, pk := range t.PKColumns {
+			pkc = append(pkc, int64(pk))
+		}
+		table.PKColumns = pkc
+		tables[table.Name] = table
+	}
+	h.lastSchema = tables
+	return nil
+}
+
+// GetTableForPos returns a best-effort schema for a specific gtid
+func (h *HistorianSvc) GetTableForPos(tableName sqlparser.TableIdent, gtid string) *binlogdatapb.MinimalTable {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	log.Infof("GetTableForPos called for %s with pos %s", tableName, gtid)
```

Comment:
These info logs will likely generate too much spam. The general guideline is to use info logs when services come up or go down, and then print important events we know for sure will be rare, like a settings change.

We could change it to `log.V` instead, which allows us to turn it on as needed: https://github.com/golang/glog. Same comment for all the other Info logs also.